### PR TITLE
Help text additions

### PR DESCRIPTION
**Pull request by Cyclam**
Some additional help text as required by the CLI team.

See checklist at the bottom of this doc for what's required. Note the TOC stuff mentioned there is (apparently) already covered, so this PR only covers the long summary, parameters and examples.

### CODE REVIEW

File: src/aosm/azext_aosm/_help.py [hunk 1]
```diff
@@ -10,60 +10,171 @@
...
     short-summary: Generate configuration file for building an AOSM publisher Network Function Definition.
+    long-summary: |
+      Generates a configuration file that you can use to build an AOSM Network Function Definition (NFD). The configuration file is a JSONC file that contains the required parameters for building the NFD. You must complete the configuration file with your specific values before building the NFD.
+    parameters:
+      - name: --definition-type
+        type: string
+        short-summary: Type of network function definition to be built.
+        long-summary: |
+          The type of AOSM definition to be built. The generate config file differs depending on type. This can be one of the following values:
+          - vnf: Virtual Network Function
```
Comment (jordlay) [inline src/aosm/azext_aosm/_help.py:39]: Maybe say for VNF "on Azure Core" like you have for the other

File: src/aosm/azext_aosm/_help.py [hunk 2]
```diff
@@ -10,60 +10,171 @@
...
+          - cnf: Container Network Function
+          - vnf-nexus: Virtual Network Function on Nexus
+      - name: --output-file
+        type: string
+        short-summary: The name of the output file to write the generated config text to.
```
Comment (jordlay) [inline src/aosm/azext_aosm/_help.py:44]: Should we be adding required and optional in the summary? it might be automatically done elsewhere tbh

Comment (Cyclam) [reply src/aosm/azext_aosm/_help.py:44]: Added.

File: src/aosm/azext_aosm/_help.py (hunk 1 above)
Comment (Cyclam) [reply src/aosm/azext_aosm/_help.py:39]: Done.

**Pull request merged**